modAionImpl/src/org/aion/zero/impl/cli/Cli.java

```diff
@@ -231,6 +231,20 @@ public ReturnType call(final String[] args, Cfg cfg) {
 
             // 5. options that can be influenced by the -d and -n arguments
 
+            if (options.getPort() != null) {
+                int port_number = Integer.parseInt(options.getPort());
```

Comment:
`parseInt` throws an exception when given a string that is not a number. This should be handled without crashing the kernel. For example `./aion.sh -p test` will show a NumberFormatException instead of properly notifying the user that an incorrect value has been given as parameter.

To solve this issue you can surround the statement with a `try-catch` and print the adequate message inside the catch, or set a boolean flag to true and check for the flag in the if statement below.

Reply:
I've noticed a tangentially-related issue that I think we should fix in this review.  Linux can be configured to require root permissions for opening certain ports.  By default, ports <= 1024 are need root, but it could be configured to be anything.  Example:

```
sergiu@aion-XPS-8930:~/repos/aion2$ ./aion.sh -p 140
[...]
18-12-05 14:28:41.107 INFO  GEN  [main]: loaded block <num=24, root=2c0448b8... l=32>
18-12-05 14:28:41.171 ERROR P2P  [main]: tcp-server-socket-exception.
java.net.SocketException: Permission denied
        at java.base/sun.nio.ch.Net.bind0(Native Method)
        at java.base/sun.nio.ch.Net.bind(Net.java:461)
        at java.base/sun.nio.ch.Net.bind(Net.java:453)
        at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
        at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:80)
        at org.aion.p2p.impl1.P2pMgr.run(P2pMgr.java:194)
        at org.aion.zero.impl.AionHub.initializeHub(AionHub.java:205)
        at org.aion.zero.impl.AionHub.<init>(AionHub.java:125)
        at org.aion.zero.impl.blockchain.AionImpl.<init>(AionImpl.java:77)
        at org.aion.zero.impl.blockchain.AionImpl.<init>(AionImpl.java:55)
        at org.aion.zero.impl.blockchain.AionImpl$Holder.<clinit>(AionImpl.java:68)
        at org.aion.zero.impl.blockchain.AionImpl.inst(AionImpl.java:72)
        at org.aion.zero.impl.blockchain.AionFactory.create(AionFactory.java:34)
        at org.aion.Aion.main(Aion.java:223)
18-12-05 14:28:41.175 INFO  GEN  [main]: <node-started endpoint=p2p://32b3fee1-a64d-4b05-baca-0f7101394b63@0.0.0.0:140>
18-12-05 14:28:41.178 INFO  CONS [main]: <sealing-disabled>
```

Then the kernel continues running.  I don't think the input validation of our code should check for this (there's other things we can't anticipate i.e. they specified a port that's already in use).  Instead, can we put a try/catch around the code that opens the port and print out an error if the port can't be opened?